src/routes/jira/connect/enterprise/app/jira-connect-enterprise-app-put.ts

```diff
@@ -1,17 +1,27 @@
-import { NextFunction, Request, Response } from "express";
+import { Request, Response, NextFunction } from "express";
+import { GitHubServerApp } from "~/src/models/github-server-app";
 
 export const JiraConnectEnterpriseAppPut = async (
 	req: Request,
 	res: Response,
 	next: NextFunction
 ): Promise<void> => {
+	req.log.debug("Received Jira Connect Enterprise App PUT request to update app.");
+
 	try {
-		req.log.debug("Received Jira Connect Enterprise App PUT request");
+		const verifiedApp = await GitHubServerApp.getForUuidAndInstallationId(req.params.uuid, res.locals.installation.id);
```

Comment:
Added this additional check to make sure the uuid in the url matches the installationId. If it doesn't match, the request fails.